### PR TITLE
Single project route authorization

### PR DESCRIPTION
**Pull request by Tom-Bonnike**

### CODE REVIEW

File: src/pages/project.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import throw404 from '@helpers/throw404'
 import Wrapper from '@components/Wrapper'
 import Sidebar from '@components/Sidebar'
 import ProjectSidebar from '@components/ProjectSidebar'
 import KeyList from '@components/KeyList'
 import KeysFilters from '@components/KeysFilters'
 import withAuthentication from '@higherOrders/withAuthentication'
 
+const validateProjectOwnership = (context: any = {}) => {
+  if (!context.query.id) {
```
Comment (Tom-Bonnike) [inline src/pages/project.js]: We should use the slug here. For the query as well.

File: src/containers/SingleProjectPage/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  return (
+    <Query query={query} variables={{ projectSlug: props.query.projectSlug }}>
+      {({ error, loading, data }) => {
+        if (error) {
+          return <ErrorPage statusCode={404} />
+        }
+
+        if (loading) {
+          return 'Loading…'
```
Comment (Tom-Bonnike) [inline src/containers/SingleProjectPage/index.js]: I’ll use the `<Loader>` component when it’s merged.

File: src/components/SingleProjectPage/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const SingleProjectPage = () => (
+  <Wrapper flex>
+    <Sidebar />
+    <ProjectSidebar />
+    <Wrapper padding="xlarge" flex align="column">
+      <KeysFilters />
+      <KeyList />
+    </Wrapper>
+  </Wrapper>
```
Comment (Tom-Bonnike) [inline src/components/SingleProjectPage/index.js]: I’ll template stuff in another PR.

File: src/higherOrders/withAuthentication/index.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ const withAuthentication = (Component: any) => {
         redirectTo(context, '/login')
       }
 
-      return {}
+      return { query: context.query }
```
Comment (Tom-Bonnike) [inline src/higherOrders/withAuthentication/index.js]: This doesn’t really belong here but yolo.

Comment (Tom-Bonnike) [reply src/higherOrders/withAuthentication/index.js]: Actually, I think I’ll refactor that using the context API at some point.

File: src/containers/SingleProjectPageContainer/index.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+type Props = {
+  projectSlug: string
+}
+
+const SingleProjectPageContainer = (props: Props) => (
+  <Wrapper flex contentCentered>
+    <Query query={query} variables={{ projectSlug: props.projectSlug }}>
+      {({ error, loading, data }) => {
+        if (error) {
+          return <ErrorPage statusCode={404} />
```
Comment (Tom-Bonnike) [inline src/containers/SingleProjectPageContainer/index.js]: Downside is we don’t render an actual 404… there’s still the sidebar 🤔

File: src/components/SingleProjectPage/index.js (hunk 3 above)
Comment (rmnbrd) [reply src/components/SingleProjectPage/index.js]: 👍

File: src/containers/SingleProjectPageContainer/index.js (hunk 5 above)
Comment (rmnbrd) [reply src/containers/SingleProjectPageContainer/index.js]: Little bit weird but... why not 🤷‍♂️

Comment (Tom-Bonnike) [reply src/containers/SingleProjectPageContainer/index.js]: I have an idea to improve that in another PR :)

**Pull request merged**